### DISCUSSION TITLE
Tasmota IR Control for Toshiba Aircondition turns AC always off

### DISCUSSION THREAD

**Discussion by MichaelUray** [arendst/Tasmota#15775] [Q&A] [answered]
Hi guys,

i  have a Perle IR WiFi adapter, also known as [auvisio S06 IR Controller](https://templates.blakadder.com/auvisio_S06.html) flashed with tasmota-ir to control a Toshiba air condition.

My active setting is `{"NAME":"IR-Transreceiver","GPIO":[0,0,0,0,52,51,0,0,0,0,8,0,0],"FLAG":0,"BASE":18}`.

I received the following from the IR remote control to setup 24°C:
```
00:34:03.030 RSL: RESULT = {"IrReceived":{"Protocol":"TOSHIBA_AC","Bits":72,"Data":"0xF20D03FC0170000071","Repeat":0,"IRHVAC":{"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}}
```

After that I created from the received information the IRhvac command to send it via the Tasmota device to the aircondition:
```
00:34:27.514 CMD: IRhvac {"Protocol":"TOSHIBA_AC","Bits":72,"Data":"0xF20D03FC0170000071","Repeat":0,"IRHVAC":{"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}
00:34:27.974 RSL: RESULT = {"IRHVAC":{"Vendor":"TOSHIBA_AC","Model":1,"Mode":"Auto","Power":"Off","Celsius":"On","Temp":21,"FanSpeed":"Medium","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}
```

The interesting thing is, that it turns the aircondition off instead of changing the setpoint.
I tried a couple commands, but it always ends with a powered off AC.
Have you guys any idea what could cause that?

I also read a couple other commands from the IR remote for testing with two different setpoints as well as turning it off and it looks like this:
```
00:01:42.035 RSL: RESULT = {"IrReceived":{"Protocol":"TOSHIBA_AC","Bits":72,"Data":"0xF20D03FC0170000071","Repeat":0,"IRHVAC":{"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}}
00:01:45.657 RSL: RESULT = {"IrReceived":{"Protocol":"TOSHIBA_AC","Bits":72,"Data":"0xF20D03FC0160000061","Repeat":0,"IRHVAC":{"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"On","Celsius":"On","Temp":23,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}}
00:06:04.722 RSL: RESULT = {"IrReceived":{"Protocol":"TOSHIBA_AC","Bits":72,"Data":"0xF20D03FC0160070066","Repeat":0,"IRHVAC":{"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"Off","Celsius":"On","Temp":23,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}}
```

**Comment by sfromis** [comment, marked as answer]
I'm not using IRHVAC, but it seems that `Data` or other fields not part of the `IRHVAC` object level in the received JSON are **not** among the suggested fields when issuing commands, as the other parameters are (I think) combined to form the data package. Try omitting those, only basing your commands on the `IRHVAC` part of the `IrReceived` payload.

**Reply by MichaelUray** [reply]
You mean like this?
```
IRhvac {"Protocol":"TOSHIBA_AC","IRHVAC":{"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}}
```

**Reply by sfromis** [reply]
No.

> only basing your commands on the IRHVAC part of the IrReceived payload

"Only" was to be taken literally, the starting `{` from that part is where you start. Docs (did you even read that?) [says](https://tasmota.github.io/docs/Tasmota-IR/#sending-irhvac-commands) that you can use the keyword "Protocol" instead of "Vendor" when replicating that part, but basically you can just copy/paste.

Again, I'm not using IRHVAC, but my assumption would be this as a starting point
```
IRhvac {"Vendor":"TOSHIBA_AC","Model":-1,"Mode":"Auto","Power":"On","Celsius":"On","Temp":24,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1}
```

**Reply by s-hadinger** [reply]
`IRhvac` shouldn't have a JSON sub-structure `IRHVAC`. Put all parameters at the same level in the JSON structure like shown by @sfromis 

I understand it does not quite match the received structure, but makes parsing easier.

**Reply by MichaelUray** [reply]
Thank you very much guys, It looks I did not read the manual good enough.
sfromis proposal works fine my Toshiba air condition. Thanks again.
